### PR TITLE
Update generic methods docs to use the new syntax.

### PR DESCRIPTION
**Pull request by munificent**
- Move the old doc to a separate file and link to it.
- Update the new doc to use the new syntax.

I also took the liberty of trying to make it a little shorter and use
more concrete examples. I hope I still cover the same bases. If not,
let me know what I missed.

### CODE REVIEW

File: pkg/dev_compiler/doc/GENERIC_METHODS.md [hunk 1]
```diff
@@ -1,229 +1,164 @@
...
-This is a summary of the current (as of January 2016) comment-based generic
-method syntax supported by the analyzer strong mode and the Dart Dev Compiler.
-The comment-based syntax essentially uses the proposed actual generic method
-syntax, but wraps it in comments.  This allows developers to experiment with
-generic methods while still ensuring that their code runs on all platforms while
-generic methods are still being evaluated for inclusion into the language.
+[proposal]: https://github.com/leafpetersen/dep-generic-methods/blob/master/proposal.md
 
-## Declaring generic method parameters
+When they were still being prototyped, and [older comment-based syntax was
```
Comment (leafpetersen) [inline pkg/dev_compiler/doc/GENERIC_METHODS.md]: and -> an

File: pkg/dev_compiler/doc/GENERIC_METHODS.md [hunk 2]
```diff
@@ -1,229 +1,164 @@
...
 
-The previous examples declared generic method parameters, but did not use them.
-You can use a generic method parameter `T` anywhere that a type is expected in
-Dart by writing a type followed by `/*=T*/`.  So for example, `dynamic /*=T*/`
-will be interpreted as `dynamic` by all non-strong mode tools, but will be
-interpreted as `T` by strong mode.  In places where it is valid to leave off a
-type, simply writing `/*=T*/` will be interpreted as `dynamic` by non-strong
-mode tools, but will be interpreted as `T` by strong mode.  For example:
+You've seen some examples already, but you can use a generic type parameter
+anywhere you would expect in a generic method.
```
Comment (leafpetersen) [inline pkg/dev_compiler/doc/GENERIC_METHODS.md]: We should probably mention that it is currently an error to use it in an `is` check?

File: pkg/dev_compiler/doc/GENERIC_METHODS.md [hunk 3]
```diff
@@ -1,229 +1,164 @@
...
-}
+var fruits = ["apple", "banana", "cherry"];
+var lengths = fruits.map((fruit) => fruit.length);
 ```
 
-## Instantiating generic classes with generic method parameters
-
-You can use generic method parameters to instantiate generic classes using the
-same `/*=T*/` syntax.
+`Iterable.map<S>(S transform(T` is now a generic method that takes a type
```
Comment (leafpetersen) [inline pkg/dev_compiler/doc/GENERIC_METHODS.md]: Missing the end of the .map signature?

File: pkg/dev_compiler/doc/GENERIC_METHODS.md (hunk 1 above)
Comment (munificent) [inline pkg/dev_compiler/doc/GENERIC_METHODS.md]: Done.

File: pkg/dev_compiler/doc/GENERIC_METHODS.md (hunk 3 above)
Comment (munificent) [inline pkg/dev_compiler/doc/GENERIC_METHODS.md]: Oops, moved it to a standalone code snippet and forgot to fix the prose. Done.

**Comment by munificent** [comment]
Thanks!

**Review by leafpetersen — APPROVED** [review]
lgtm

**Pull request merged**